### PR TITLE
Provided the nbs processing status for the hl7 files that have been u…

### PR DESCRIPTION
**Pull request by ssathiah**
Provided the nbs processing status for the hl7 files that have been uploaded and processed.

### CODE REVIEW

File: data-ingestion-service/src/main/java/gov/cdc/dataingestion/camel/routes/ElrProcessStatusComponent.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    public String process(String body){
+        String status = "";
+            logger.debug("File name and Elr Id:{}", body);
+            if (body != null && !body.trim().isEmpty()) {
+                //body = "HL7file-sftpstatus1.txt:7DAC34BD-B011-469A-BF27-25904370E9E3";//NOSONAR
+                String[] fileNameElr = body.split(":");
+                String elrId = fileNameElr[1].trim();
+                MessageStatus messageStatus = reportStatusService.getMessageStatus(elrId);
+                if (messageStatus.getNbsInfo().getNbsInterfaceStatus() != null && messageStatus.getNbsInfo().getNbsInterfaceStatus().equals(SUCCESS)) {
+                    status = SUCCESS;
```
Comment (ndduc01) [inline data-ingestion-service/src/main/java/gov/cdc/dataingestion/camel/routes/ElrProcessStatusComponent.java:42]: Should we also append ELR ID when then message is success?

File: data-ingestion-service/src/main/java/gov/cdc/dataingestion/odse/repository/IEdxActivityLogRepository.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+@Repository
+public interface IEdxActivityLogRepository extends JpaRepository<EdxActivityDetailLog, Long> {
+    @Query(value = """
+            select eadl.record_id as recordId ,eadl.record_type as recordType ,eadl.log_type as logType,eadl.log_comment as logComment,
+            eal.record_status_time as recordStatusTime from NBS_ODSE.dbo.EDX_activity_detail_log eadl,NBS_ODSE.dbo.EDX_activity_log eal 
+            where eal.source_uid = :sourceId and eadl.edx_activity_log_uid =eal.edx_activity_log_uid
+            """,
+            nativeQuery = true)
+    List<EdxActivityLogModelView> getEdxActivityLogDetailsBySourceId(Long sourceId);
```
Comment (ndduc01) [inline data-ingestion-service/src/main/java/gov/cdc/dataingestion/odse/repository/IEdxActivityLogRepository.java:19]: do you need to add @param Annotation here like this, I might work without the annotation just double check
@Param("uid") Long uid

File: data-ingestion-service/src/main/java/gov/cdc/dataingestion/camel/routes/ElrProcessStatusComponent.java (hunk 1 above)
Comment (ssathiah) [reply data-ingestion-service/src/main/java/gov/cdc/dataingestion/camel/routes/ElrProcessStatusComponent.java:42]: Anil said a 'Success' message is enough when the record has been processed successfully.

**Review by ndduc01 — APPROVED** [review]
Looks good

File: data-ingestion-service/src/main/java/gov/cdc/dataingestion/odse/repository/IEdxActivityLogRepository.java (hunk 2 above)
Comment (ssathiah) [reply data-ingestion-service/src/main/java/gov/cdc/dataingestion/odse/repository/IEdxActivityLogRepository.java]: Added @Param

**Pull request merged**